src/content/getusermedia/getdisplaymedia/js/main.js

```diff
@@ -0,0 +1,39 @@
+/*
+ *  Copyright (c) 2018 The WebRTC project authors. All Rights Reserved.
+ *
+ *  Use of this source code is governed by a BSD-style license
+ *  that can be found in the LICENSE file in the root of the source
+ *  tree.
+ */
+'use strict';
+
+// Polyfill in Firefox.
+// See https://blog.mozilla.org/webrtc/getdisplaymedia-now-available-in-adapter-js/
+if (adapter.browserDetails.browser == 'firefox') {
+  adapter.browserShim.shimGetDisplayMedia(window, 'screen');
+}
+
+function handleSuccess(stream) {
+  const video = document.querySelector('video');
+  video.srcObject = stream;
+}
+
+function handleError(error) {
+  errorMsg(`getDisplayMedia error: ${error.name}`, error);
+}
+
+function errorMsg(msg, error) {
+  const errorElement = document.querySelector('#errorMsg');
+  errorElement.innerHTML += `<p>${msg}</p>`;
+  if (typeof error !== 'undefined') {
+    console.error(error);
+  }
+}
+
+if ('getDisplayMedia' in navigator) {
+  navigator.getDisplayMedia({video: true})
```

Comment:
Extract the configuration to a separate global object. Since most screen sharing will be about sharing a specific window/application from the desktop, I suggest we do that instead of sharing the entire screen.

Reply:
getDisplayMedia currently doesn't support constraints.